### PR TITLE
Change embeddings

### PR DESCRIPTION
**Pull request by reeceshuttle**
This pull request changes how we use embeddings to support using both the init and goal strings. 
Do we want to use flags to indicate if we only want to use the init string vs goal string?

### CODE REVIEW

**Comment by tomsilver** [comment]
>Do we want to use flags to indicate if we only want to use the init string vs goal string?

your call! I'm fine leaving this for the future, if ever

File: llm4pddl/approaches/llm_open_loop_approach.py [hunk 1]
```diff
@@ -203,14 +204,17 @@ def _llm_response_to_plan(response: LLMResponse, task: Task) -> Plan:
...
-    def _embed_task(self, task: Task) -> Embedding:
+    def _embed_task(self, task: Task) -> Dict[str, Embedding]:
         """Embeds a task using embedding_model."""
         # note: task_string includes the Q: and A: parts, not just task string
-        task_string = self._create_prompt(task)
-        embedding = self._embedding_model.encode(task_string)
-        return embedding
+        init_str = utils.get_init_str(task)
+        goal_str = utils.get_goal_str(task)
+        # task_string = self._create_prompt(task)
```
Comment (tomsilver) [inline llm4pddl/approaches/llm_open_loop_approach.py:212]: remove

File: llm4pddl/approaches/llm_open_loop_approach.py [hunk 2]
```diff
@@ -234,17 +239,43 @@ def _get_closest_datums(self, task: Task,
...
         ]
-        # we will need a get_init_string() and get_goal_string() helper funcs.
-        indices = np.argsort(cos_sims)[-num_closest:]
+        goal_cos_sims = np.array([
+            self._get_cosine_sim(task_goal_embedding, other_emb)
+            for other_emb in other_goal_embeddings
+        ])
+        total_cos_sims += goal_cos_sims
+
+        # # now compare this embedding to all the other embeddings
```
Comment (tomsilver) [inline llm4pddl/approaches/llm_open_loop_approach.py:269]: remove

**Review by tomsilver — APPROVED** [review]
excellent! just a few small things

**Comment by reeceshuttle** [comment]
> 

sounds good Im just going to leave it as is for now. We can make flags for either one if we want later.

**Comment by reeceshuttle** [comment]
Done!

**Pull request merged**